apps/api/src/user/service/user.service.ts

```diff
@@ -51,6 +82,29 @@ export class UserService {
       isActive: dto.isActive,
       isOnboardingFinished: dto.isOnboardingFinished
     }
+
+    if (dto.email) {
+      const userwithEmail = await this.prisma.user.findFirst({
+        where: {
+          email: dto.email
+        }
+      })
+
+      if (userwithEmail) {
+        throw new ConflictException('User with this email already exists')
+      }
+
+      const otp = await this.prisma.userEmailChange.create({
+        data: {
+          newEmail: dto.email,
+          userId: userId,
+          otp: randomUUID().slice(0, 6).toUpperCase()
+        }
+      })
+
+      await this.mailService.sendOtp(dto.email, otp.otp)
```

Comment:
You don't need to do this if the admin is updating the user's email.

Reply:
Removed sending otp for email changes done by admin